### PR TITLE
HTML to Markdown Final PR

### PR DESCRIPTION
**Pull request by mellenberger**

### CODE REVIEW

File: client/static/css/CodeChatEditor.css [hunk 1]
```diff
@@ -207,3 +207,13 @@ body {
     margin-bottom: 0px;
     padding-bottom: 0px;
 }
+
+.CodeChat-TinyMCE-TableDefault {
```
Comment (bjones1) [inline client/static/css/CodeChatEditor.css:211]: Add a comment with a hyperlink back to the `visual_table_class` in `tinymce-webpack.mts` to help explain where it's used.

File: client/src/tinymce-webpack.mts [hunk 2]
```diff
@@ -1,48 +1,41 @@
...
-//         General Public License for more details.</p>
-//     <p>You should have received a copy of the GNU General Public License
-//         along with the CodeChat Editor. If not, see <a
-//             href="http://www.gnu.org/licenses/">http://www.gnu.org/licenses/</a>.
-//     </p>
-// </details>
-// <h1><code>tinymce-webpack.ts</code> &mdash; imports the TinyMCE editor from
-//     NPM packages using webpack</h1>
-// <p>Import TinyMCE.</p>
+// Copyright (C) 2022 Bryan A. Jones.
```
Comment (bjones1) [inline client/src/tinymce-webpack.mts:1]: Does the markdown-to-html process exclude the `<summary>` and `<detail>` tags?

**Review by bjones1 — CHANGES_REQUESTED** [review]

File: client/src/tinymce-webpack.mts (hunk 2 above)
Comment (mellenberger) [reply client/src/tinymce-webpack.mts:1]: Yes, it removes these tags & doesn't have a conversion for them

File: client/static/css/CodeChatEditor.css (hunk 1 above)
Comment (mellenberger) [reply client/static/css/CodeChatEditor.css:184]: added in [06e78c7](https://github.com/bjones1/CodeChat_Editor/pull/24/commits/06e78c728b3b0cbeab4203e7ee907525d4d26e78)

**Comment by mellenberger** [comment]
I added the method for adding custom classes. Currently, only subscript & superscript has been added with these custom rules. It is more of a proof of concept at the moment due to only being able to add single-character markdown conversions within these new rules.

File: client/src/tinymce-webpack.mts (hunk 2 above)
Comment (bjones1) [reply client/src/tinymce-webpack.mts:1]: OK, thanks. I'll work on this in a separate sprint.

**Comment by bjones1** [comment]
Nice!

File: client/static/css/CodeChatEditor.css [hunk 3]
```diff
@@ -42,138 +38,115 @@ html {
...
-    <li><code>&lt;div class="CodeChat-doc-indent"&gt;</code> contains
-        &lt;doc block indent&gt;</li>
-    <li><code>&lt;div class="CodeChat-TinyMCE"&gt;</code> contains the
-        &lt;doc block contents&gt;</li>
-</ul> */
+    /* `` `Use flexbox layout to style doc blocks. The approach is similar to code blocks; the overall goal is:` ``
+
+where:
+
+*   `<div class="CodeChat-ACE-gutter-padding">` contains
```
Comment (bjones1) [inline client/static/css/CodeChatEditor.css:111]: Lots more mangling there...

File: client/tsconfig.json [hunk 4]
```diff
@@ -36,7 +36,8 @@
         "outDir": "out",
         "rootDir": "src",
         "strict": true,
-        "target": "ES2022"
+        "target": "ES2022",
+        "resolveJsonModule": true
```
Comment (bjones1) [inline client/tsconfig.json:40]: Explain this (needed for importing the JSON file).

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Comment by ClarkHensley** [comment]
I'd like to thank Github for not notifying me of these comments.

It looks like the anchor tag and the smart dash are issues.

However, I just pulled down a clean copy of the merge (testing it on my desktop as opposed to my laptop), and I'm seeing none of the issues in the .css file. I checked all the issues, and everything there is working exactly as you're requesting.

**Comment by ClarkHensley** [comment]
Hmm, not none of the issues. It is still destroying the <max line numbers space><code> section. And adding too many bullet points to the list.

But I'm not getting any extra backticks

**Comment by ClarkHensley** [comment]
I'll stop flooding the comments. Kenzie showed me the issue, I can now replicate it

**Comment by ClarkHensley** [comment]
(Hopefully this just adds onto the PR like I want)

So, this version should fix most of the issues shown in the .css file.
In essence, it correctly converts "<" and ">" to "<" and ">" respectively.

The main issue:  \<code\> tags render these escaped html entities as code, so they remain escaped.
We'll have to explore adding in a specific rule for  tags, but that should be a very specific corner-case.

Otherwise, I do believe this solves those other issues with "<" and ">" mangling, the stray "`" characters, etc.

**Comment by bjones1** [comment]
Hmm, this looks tricky to solve for all cases -- I'll merge this as a good start. Thanks!

**Pull request merged**